pontoon/base/static/js/translate.js

```diff
@@ -1995,98 +1996,77 @@ var Pontoon = (function (my) {
-                        id: entity.id
-                      });
-                    }
-
-                    self.updateTranslation(entity, pluralForm, {
-                      pk: next.data('id'),
-                      string: newTranslation,
-                      approved: false,
-                      fuzzy: false
-                    });
-
-                  // Last translation deleted, no alternative available
-                  } else {
-                    if (self.cachedTranslation === $('#translation').val()) {
-                      $('#clear').click();
-                    }
-
-                    if (entity.body && pluralForm === 0) {
-                      self.postMessage("DELETE", {
-                        id: entity.id
-                      });
-                    }
-
-                    self.updateTranslation(entity, pluralForm, {
-                      pk: null,
-                      string: null,
-                      approved: false,
-                      fuzzy: false
-                    });
-
-                    $('#helpers .history ul')
-                      .append('<li class="disabled">' +
-                                '<p>No translations available.</p>' +
-                              '</li>');
-                  }
-
-                  self.moveCursorToBeginning();
-                  self.updateCurrentTranslationLength();
-                  self.updateCachedTranslation();
-                  self.updateFilterUI();
-                }
+            self.updateTranslation(entity, pf, data.translation);
 
-                // Update number of history entities
-                var count = $('#helpers .history .suggestion').length;
-                $('#helpers a[href="#history"] .count')
-                  .toggle(count > 0)
-                  .html(count);
-              });
+            var item = button.parents('li');
+            item.addClass('rejected').removeClass('translated');
+            item.find('.unapprove').removeClass('unapprove').addClass('approve').prop('title', 'Approve');
+            button.addClass('unreject').removeClass('reject').prop('title', 'Unreject');
           },
           error: function() {
             self.endLoader('Oops, something went wrong.', 'error');
           }
         });
       });
 
+      $('#helpers .history').on('click', 'menu .unreject', function (e) {
+         var button = $(this),
+             translationId = parseInt($(this).parents('li').data('id'));
+
+         $.post('/unreject-translation/', {
+            csrfmiddlewaretoken: $('#server').data('csrf'),
+            translation: translationId,
+            paths: self.getPartPaths(self.currentPart)
+         }).then(function(data) {
+           var entity = self.getEditorEntity();
+           var pf = self.getPluralForm(true);
+
+           self.stats = data.stats;
+
+           self.updateTranslation(entity, pf, data.translation);
+
+           $('#translation').val(data.translation.string).focus();
```

Comment:
In case #651 gets merged earlier, we should replace this with a function. :-)
https://github.com/mozilla/pontoon/pull/651/files#diff-5ec2d12043f59ff839712736b821ec71L458